internal/interfaces.go

```diff
@@ -17,12 +17,12 @@ type Writer interface {
 	WriteConfig(ctx context.Context, cfg Config) error
 
 	WriteStats(ctx context.Context, fs FileSystem) error
-	WriteMatchesSummary(ctx context.Context, fs FileSystem) error
+	WriteMatchesSummary(ctx context.Context, fs FileSystem, pocEnabled bool) error
```

Comment:
It looks like this option is specific to the `writer.JSON` implementation (I see it's not used by the rest of the `scan.Writer` implementations), and since it is "global" for the entire scan execution, I think we can just make it an option of this writer.

So, pass it to `writer.NewJSON` as a functional option, store it in the `writer.JSON` instance, and use it from the methods as needed, like now but from as an attribute instead of a function argument.